core/cli/src/main/java/org/elasticsearch/cli/Command.java

```diff
@@ -103,14 +94,10 @@ public final int main(String[] args, Terminal terminal) throws Exception {
     }
 
     /**
-     * Indicate whether or not logging should be configured without reading a log4j2.properties. Most commands should do this because we do
-     * not configure logging for CLI tools. Only commands that configure logging on their own should not do this.
-     *
-     * @return true if logging should be configured without reading a log4j2.properties file
+     * Setup method to be executed before parsing or execution of the command being run. Any exceptions thrown by the
+     * method will not be cleanly caught by the parser.
      */
-    protected boolean shouldConfigureLoggingWithoutConfig() {
-        return true;
-    }
+    protected void beforeExecute() throws Exception {}
```

Comment:
Why does this declare a checked exception?

Reply:
cruft from the refactor. itll be gone next push.